internal/prefixcollector/collector.go

```diff
@@ -23,113 +23,66 @@ import (
 
 	"github.com/networkservicemesh/sdk/pkg/tools/spanhelper"
 
-	apiV1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/watch"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
-
-	"github.com/ghodss/yaml"
-
 	"github.com/sirupsen/logrus"
 
 	"github.com/networkservicemesh/sdk/pkg/tools/prefixpool"
 )
 
-const (
-	configMapKey = "excluded_prefixes.yaml"
-)
-
-// ExcludePrefixSource is source of excluded prefixes
-type ExcludePrefixSource interface {
+// PrefixSource is source of excluded prefixes
+type PrefixSource interface {
 	Prefixes() []string
 }
 
-// ExcludePrefixCollector is service, collecting excluded prefixes from list of ExcludePrefixSource
+// WritePrefixesFunc is excluded prefixes write func
+type WritePrefixesFunc func(context.Context, []string)
+
+// WatchPrefixesFunc is excluded prefixes resource watch func
+type WatchPrefixesFunc func(context.Context, *utils.SynchronizedPrefixesContainer)
+
+// ExcludePrefixCollector is service, collecting excluded prefixes from list of PrefixSource
 // and writing result to outputFilePath in yaml format
 type ExcludePrefixCollector struct {
-	notifyChan            <-chan struct{}
-	nsmConfigMapName      string
-	nsmConfigMapNamespace string
-	sources               []ExcludePrefixSource
-	previousPrefixes      *utils.SynchronizedPrefixesContainer
-	configMapInterface    v1.ConfigMapInterface
+	notifyChan       <-chan struct{}
+	writeFunc        WritePrefixesFunc
+	watchFunc        WatchPrefixesFunc
+	sources          []PrefixSource
+	previousPrefixes *utils.SynchronizedPrefixesContainer
 }
 
 // NewExcludePrefixCollector creates ExcludePrefixCollector
-func NewExcludePrefixCollector(notifyChan <-chan struct{}, configMapName, configMapNamespace string,
-	sources ...ExcludePrefixSource) *ExcludePrefixCollector {
-	return &ExcludePrefixCollector{
-		nsmConfigMapName:      configMapName,
-		nsmConfigMapNamespace: configMapNamespace,
-		notifyChan:            notifyChan,
-		sources:               sources,
-		previousPrefixes:      utils.NewSynchronizedPrefixesContainer(),
+func NewExcludePrefixCollector(notifyChan <-chan struct{}, writeFunc WritePrefixesFunc,
+	watchFunc WatchPrefixesFunc, sources ...PrefixSource) *ExcludePrefixCollector {
```

Comment:
I still think that `options` here can make code some clear.

For example:

```suggestion
func NewExcludePrefixCollector(options...Option) *ExcludePrefixCollector {
```

Then main.go could look something like:

```go
func main() {
    ...
    var options []Option
    if config.OutputFilePath != "" {
        options = append(options, WithOutputFile(config.OutputFilePath))
    }
    ...
    go NewExcludePrefixCollector(options...).Serve(ctx)
    <-ctx.Done()
}
```

Reply:
Added options for collector